### ISSUE TITLE
**tl;dr** No.

### ISSUE THREAD

**Issue by mbatiuk** [MarioniLab/DropletUtils#60]
**tl;dr** No.

The long answer is that you can use `removeSwappedDrops()` to run the same algorithm on any molecule-level information, regardless of format. The emphasis here is _molecule-level_; you need the UMI, assigned gene and assigned cell for each individual transcript molecule. This is what is returned by Cellranger in their molecule information HDF5 file.

It is not sufficient to run this on the count matrix, which aggregates the molecules into counts at the gene level. This discards information about individual molecules, preventing `swappedDrops()` from making a decision about whether a particular molecule is a swapping artifact. I'm not familiar enough with Starsolo to know whether molecule-level stats are generated.

_Originally posted by @LTLA in https://github.com/MarioniLab/DropletUtils/issues/59#issuecomment-780178681_
